src/components/Tabs/Annotation/AnnotationImportButton.tsx

```diff
@@ -0,0 +1,139 @@
+        setErrorText('Could not parse CSV file. Parsing failed with the following error: "' + error + '"');
+      }
+    };
+    reader.readAsText(file);
+  };
+
+  const handleCancel = (): void => {
+    setShowModal(false);
+    setConvertedAnnotationData(null);
+    setUploadedFile(null);
+  };
+
+  const handleImport = async (): Promise<void> => {
+    if (convertedAnnotationData) {
+      // TODO: give more advanced merging options here. There are three possible
+      // options:
+      // 1. Overwrite existing annotations (default, current behavior)
+      // 2. Keep both (no merging, labels are kept separate even if they have
+      //    matching names)
+      // 3. Merge annotations (merge matching labels with the same types. Users
+      //    should be given an option for how to handle when conflicts occur for
+      //    values (e.g. the imported CSV has a different value assigned to the
+      //    same ID))
+      annotationState.replaceAnnotationData(convertedAnnotationData);
+      setShowModal(false);
+      setConvertedAnnotationData(null);
+      setUploadedFile(null);
+      setErrorText("");
+      annotationState.setVisibility(true);
+    }
+  };
+
+  const hasAnnotationData = annotationState.data.getLabels().length > 0;
+
+  const fileList: UploadFile[] = [];
+  if (uploadedFile) {
+    fileList.push({
+      uid: "-1",
+      name: uploadedFile.name,
+      status: convertedAnnotationData !== null ? "done" : "error",
+      // url: URL.createObjectURL(uploadedFile),
+    });
+  }
+
+  const handleFileChange = (info: { fileList: UploadFile[] }): void => {
+    if (info.fileList.length === 0) {
+      setUploadedFile(null);
+      setConvertedAnnotationData(null);
+      setErrorText("");
+    }
+  };
+
+  return (
+    <div ref={modalContainerRef}>
+      <Modal
+        getContainer={() => modalContainerRef.current!}
+        title="Import CSV"
+        open={showModal}
+        okText="Import"
+        okButtonProps={{ disabled: !convertedAnnotationData }}
+        onOk={handleImport}
+        onCancel={handleCancel}
+        destroyOnClose={true}
+      >
+        <Upload.Dragger
+          name="file"
+          multiple={false}
+          accept=".csv"
+          fileList={fileList}
+          onChange={handleFileChange}
+          showUploadList={true}
+          beforeUpload={handleFileUpload}
+        >
+          <span style={{ color: theme.color.text.hint, fontSize: theme.font.size.header }}>
+            <UploadOutlined />
+          </span>
+          <p style={{ color: theme.color.text.hint }}>Click or drag a .csv file to this area to upload</p>
+        </Upload.Dragger>
+        {errorText && <p style={{ color: theme.color.text.error }}>{errorText}</p>}
+        {convertedAnnotationData && hasAnnotationData && (
+          <p style={{ color: theme.color.text.warning }}>
+            <WarningOutlined /> Existing annotations will be overwritten during import.
+          </p>
+        )}
```

Comment:
![image](https://github.com/user-attachments/assets/307b4140-e0f9-4878-8443-920da92c38e9)

The yellow text is a bit hard to read, so this is just a placeholder for now.